packages/devtools_app/lib/src/shared/preferences/preferences.dart

```diff
@@ -69,44 +99,95 @@ class PreferencesController extends DisposableController
 
   Future<void> init() async {
     // Get the current values and listen for and write back changes.
-    final darkModeValue = await storage.getValue('ui.darkMode');
+    await _initDarkMode();
+    await _initVmDeveloperMode();
+    await _initWasmEnabled();
+    await _initVerboseLogging();
+
+    await inspector.init();
+    await memory.init();
+    await logging.init();
+    await performance.init();
+    await devToolsExtensions.init();
+
+    setGlobal(PreferencesController, this);
+  }
+
+  Future<void> _initDarkMode() async {
+    final darkModeValue =
+        await storage.getValue(_UiPreferences.darkMode.storageKey);
     final useDarkMode = (darkModeValue == null && useDarkThemeAsDefault) ||
         darkModeValue == 'true';
     ga.impression(gac.devToolsMain, gac.startingTheme(darkMode: useDarkMode));
     toggleDarkModeTheme(useDarkMode);
     addAutoDisposeListener(darkModeEnabled, () {
-      storage.setValue('ui.darkMode', '${darkModeEnabled.value}');
+      storage.setValue(
+        _UiPreferences.darkMode.storageKey,
+        '${darkModeEnabled.value}',
+      );
     });
+  }
 
+  Future<void> _initVmDeveloperMode() async {
     final vmDeveloperModeValue = await boolValueFromStorage(
-      'ui.vmDeveloperMode',
+      _UiPreferences.vmDeveloperMode.storageKey,
       defaultsTo: false,
     );
     toggleVmDeveloperMode(vmDeveloperModeValue);
     addAutoDisposeListener(vmDeveloperModeEnabled, () {
-      storage.setValue('ui.vmDeveloperMode', '${vmDeveloperModeEnabled.value}');
+      storage.setValue(
+        _UiPreferences.vmDeveloperMode.storageKey,
+        '${vmDeveloperModeEnabled.value}',
+      );
     });
+  }
 
-    await _initVerboseLogging();
+  Future<void> _initWasmEnabled() async {
+    // TODO(https://github.com/flutter/devtools/issues/7856): set the current
+    // value based on whether DevTools is actually loaded with wasm.
+    wasmMode.value = false;
```

Comment:
@eyebrowsoffire what is the best way to check from the web app itself if wasm is being used?

Reply:
You can use this variable here: https://github.com/flutter/flutter/blob/main/packages/flutter/lib/src/foundation/constants.dart#L95C12-L95C19

Reply:
`const bool.fromEnvironment('dart.library.ffi')` returns true for Wasm and false for JS

Reply:
ninja'd by @eyebrowsoffire (`kIsWasm` is better because it's official)

Reply:
Will `kIsWasm` still work if we are trying to check if the renderer is `canvaskit` or `skwasm`? Can DevTools have both a dart2js compilation and a dart2wasm compilation at runtime? It was my understanding that we'd be compiling with dart2wasm, and optionally switching between the canvaskit and skwasm renderers, but maybe I am misunderstanding what it means to "opt-in".

Reply:
As of right now, there are just two build variants available when you do dual compile: dart2wasm+skwasm and dart2js+canvaskit. dart2wasm+canvaskit is technically possible but not currently made available via our build process. As of right now, dart2wasm compilation and skwasm are a package deal, so if you select canvaskit you're getting dart2js